### PR TITLE
Replace compiler internal API with compiler main function call besides completion

### PR DESCRIPTION
**Pull request by zhelenskiy**
Second attempt for the reverted [previous PR](https://github.com/JetBrains/kotlin-compiler-server/pull/641).

### CODE REVIEW

**Review by zoobestik — APPROVED** [review]

**Comment by zoobestik** [comment]
* [ ] 1. `line` and `ch` higher than 1 for:
  ```kotlin
  fun main() {
    dfsdf
  }
  ```
  ```json5
   { /* ... */ "start": { "line": 2, "ch": 3 }, /* ... */ }
  ```
  
* [ ] 2. Not found main - `<errStream>No main method found in project.\n</errStream>`
  ```kotlin
  package org.kotlinlang.play

  fun main() {
    println("Hello, World!")
  }
  ```
* [ ] 3. none some deprecation warnings
  ```kotlin
  fun main() {
    println("abc"[0].toInt())
  }
  ```
  **Expected**:
    `toInt(): Int' is deprecated. Conversion of Char to Number is deprecated. Use Char.code property instead.`
  
* [ ] 4. no `className` for errors with severities (`WARNING`, `ERROR`):
  ```kotlin
  fun main() {
    val a: dynamic = "abc"
    println(a.charAt(1).repeat(3))     
  }
  ```
  Before:
  ```json5
  {"interval":{ /* ... */ }, "message": "..." , "severity": "ERROR", "className":"red_wavy_line" }
  ```

File: src/main/kotlin/com/compiler/server/compiler/components/CliUtils.kt [hunk 1]
```diff
@@ -16,6 +17,7 @@ import java.nio.file.Paths
 import java.util.*
 import kotlin.io.path.*
 
+fun minusOne(value: Int) = if (value > 0) value - 1 else value
```
Comment (zhelenskiy) [inline src/main/kotlin/com/compiler/server/compiler/components/CliUtils.kt:20]:
```suggestion
private fun minusOne(value: Int) = if (value > 0) value - 1 else value
```

File: src/main/kotlin/com/compiler/server/compiler/components/CliUtils.kt [hunk 2]
```diff
@@ -74,7 +76,15 @@ fun <T> CLICompiler<*>.tryCompilation(inputDirectory: Path, inputFiles: List<Pat
...
       val errorFilePath = location?.path?.let(::Path)?.outputFilePathString() ?: defaultFileName
-      val errorDescriptor = ErrorDescriptor(textInterval, message, messageSeverity, className = "WARNING".takeIf { messageSeverity == ProjectSeveriry.WARNING })
+
+      val errorDescriptor =
+        ErrorDescriptor(textInterval, message, messageSeverity, className = messageSeverity.name.let {
+          when {
+            !message.startsWith(UNRESOLVED_REFERENCE_PREFIX) && severity == ERROR -> "red_wavy_line"
+            else -> it
+          }
+        })
```
Comment (zhelenskiy) [inline src/main/kotlin/com/compiler/server/compiler/components/CliUtils.kt:86]:
```suggestion

      val className = if (!message.startsWith(UNRESOLVED_REFERENCE_PREFIX) && severity == ERROR) "red_wavy_line" else messageSeverity.name
      val errorDescriptor = ErrorDescriptor(textInterval, message, messageSeverity, className)
```

**Comment by zhelenskiy** [comment]
Is it merged?

**Pull request merged**